converters/fromarray.ts

```diff
@@ -0,0 +1,115 @@
+import { IDiscoverInsert, IEvent, IEventArray, ILayoutState } from "../declarations/clarity";
+import { Action, ObjectType, Origin } from "../declarations/clarity";
+import schemas from "./schema";
+
+export default function(eventArray: IEventArray): IEvent {
+  let id        = eventArray[0];
+  let origin    = eventArray[1];
+  let type      = eventArray[2];
+  let time      = eventArray[3];
+  let dataArray = eventArray[4];
+  let schema    = eventArray[5];
+
+  if (typeof schema === "string") {
+    schema = schemas.getSchema(schema);
+  } else {
+    schemas.addSchema(schema);
+  }
+
+  let data = dataFromArray(dataArray, schema as any[]);
+  let event: IEvent = { origin, type, id, time, data };
+  return event;
+}
+
+export function eventsFromDiscoverArray(
```

Comment:
>eventsFromDiscoverArray [](start = 16, length = 23)

Why do we handle discover separately in this new world of generic compression? Shouldn't it automatically be taken care of as it we process different schemas?